docs/architecture/adr-004-genesis-modes.md

```diff
@@ -0,0 +1,124 @@
+# ADR 4: Multiple Genesis modes
+
+## Changelog
+
+- 07/26/2019: First draft
+
+## Context
+
+### Problem
+
+Currently, some of the modules `GenesisState` are populated only if the corresponding
+object passed through the Genesis JSON is empty (_i.e_ as an empty array) or if it's
+undefined or the value is zero.
+
+This leads to confusion as you might pass a valid genesis file which contains zero/empty
+values and have `InitGenenesis` perform this unnecessary check.
+
+### Proposed Solution
+
+The proposed solution is to have two ways of initializing the daemon:
+
+1. **New (empty) chain**: the user provides the minimum amount of genesis fields required to
+start a chain. The rest is loaded directly from the mandatory genesis parameters.
+2. **Restarted chain**: the user provides all the fields from genesis.
+
+If an empty chain from `1.` receives _any_ field that should be populated, the initialization
+of the app should `panic`. Same applies for `2.` if a single field is missing.
+
+The fields that are not required are the ones in general are checked with simulation
+invariants.
+
+## Decision
+
+Add a flag `FlagPopulateGenesis = "populate-genesis"` on `server/start.go` that is
+passed to the `AppCreator` (eg: `NewSimApp` or `NewGaiaApp`) when `<app>d start` is called.
```

Comment:
As explained in my other comment, I think all field autogeneration should be an explicitly seperate process/separate command. The reason it should be a separate command is that is would be performing a fundamentally different task than `init` which initializes the chain from genesis - this auto-filling should not start the chain or initialize the db etc.

Reply:
so basically you mean creating a command to Import + update + Export

Reply:
correct - import the blank genesis file, make any updates required, then export back to the file <- all this wrapped neatly in a command